iocore/cache/Cache.cc

```diff
@@ -2000,6 +2000,12 @@ CacheProcessor::mark_storage_offline(CacheDisk *d ///< Target disk
   uint64_t total_dir_delete   = 0;
   uint64_t used_dir_delete    = 0;
 
+  /* Don't mark it again, it will invalidate the stats! */
+  if (d->offline) {
+    return this->has_online_storage();
+  }
+  d->offline = true;
```

Comment:
@gtenev and I discussed this. The problem is that in the common case, the disk is already bad when ``mark_storage_offline`` is called, so we can't depend on the good->bad state transition to know when to update the accounting.

@gtenev This looks fine to me, but I'd suggest calling the flag ``online`` so that we avoid the double negatives.